### PR TITLE
Update version to v7.7.0 and ODL dependency to v7.16.0

### PR DESCRIPTION
**Pull request by KenitoInc**
### Issues

*This pull request fixes #xxx.*

### Description

*Briefly describe the changes of this pull request.*

### Checklist (Uncheck if it is not completed)

- [ ] *Test cases added*
- [ ] *Build and test with one-click build and test script passed*

### Additional work necessary

*If documentation update is needed, please add "Docs Needed" label to the issue and provide details about the required document change in the issue.*

### CODE REVIEW

**Review by xuzhg — APPROVED** [review]

**Review by ElizabethOkerio — DISMISSED** [review]

**Review by ElizabethOkerio — APPROVED** [review]

File: src/Microsoft.AspNet.OData/Microsoft.AspNet.OData.csproj [hunk 1]
```diff
@@ -30,17 +30,14 @@
       <HintPath>..\..\sln\packages\Microsoft.Extensions.DependencyInjection.Abstractions.1.0.0\lib\netstandard1.0\Microsoft.Extensions.DependencyInjection.Abstractions.dll</HintPath>
       <Private>True</Private>
     </Reference>
-    <Reference Include="Microsoft.OData.Core, Version=7.15.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <HintPath>..\..\sln\packages\Microsoft.OData.Core.7.15.0\lib\net45\Microsoft.OData.Core.dll</HintPath>
-      <Private>True</Private>
+    <Reference Include="Microsoft.OData.Core, Version=7.16.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\..\sln\packages\Microsoft.OData.Core.7.16.0\lib\net45\Microsoft.OData.Core.dll</HintPath>
```
Comment (gathogojr) [inline src/Microsoft.AspNet.OData/Microsoft.AspNet.OData.csproj:34]: Should the 3 `<Private>True</Private>` nested elements be retained?

File: test/E2ETest/Microsoft.Test.E2E.AspNet.OData/Build.AspNet/Microsoft.Test.E2E.AspNet.OData.csproj [hunk 2]
```diff
@@ -43,17 +43,14 @@
       <HintPath>..\..\..\..\sln\packages\Microsoft.Extensions.DependencyInjection.Abstractions.1.0.0\lib\netstandard1.0\Microsoft.Extensions.DependencyInjection.Abstractions.dll</HintPath>
       <Private>True</Private>
     </Reference>
-    <Reference Include="Microsoft.OData.Client, Version=7.15.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <HintPath>..\..\..\..\sln\packages\Microsoft.OData.Client.7.15.0\lib\net45\Microsoft.OData.Client.dll</HintPath>
-      <Private>True</Private>
+    <Reference Include="Microsoft.OData.Client, Version=7.16.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\..\..\..\sln\packages\Microsoft.OData.Client.7.16.0\lib\net45\Microsoft.OData.Client.dll</HintPath>
```
Comment (gathogojr) [inline test/E2ETest/Microsoft.Test.E2E.AspNet.OData/Build.AspNet/Microsoft.Test.E2E.AspNet.OData.csproj:47]: Should the 3 `<Private>True</Private>` nested elements be retained?

File: test/E2ETest/Microsoft.Test.E2E.AspNet.OData/Build.AspNetCore/Microsoft.Test.E2E.AspNetCore.OData.csproj [hunk 3]
```diff
@@ -1858,21 +1858,17 @@
     <Reference Include="Microsoft.Net.Http.Headers, Version=2.2.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60, processorArchitecture=MSIL">
       <HintPath>..\..\..\..\sln\packages\Microsoft.Net.Http.Headers.2.2.0\lib\netstandard2.0\Microsoft.Net.Http.Headers.dll</HintPath>
     </Reference>
-    <Reference Include="Microsoft.OData.Client, Version=7.15.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <HintPath>..\..\..\..\sln\packages\Microsoft.OData.Client.7.15.0\lib\net45\Microsoft.OData.Client.dll</HintPath>
-      <Private>True</Private>
+    <Reference Include="Microsoft.OData.Client, Version=7.16.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\..\..\..\sln\packages\Microsoft.OData.Client.7.16.0\lib\net45\Microsoft.OData.Client.dll</HintPath>
```
Comment (gathogojr) [inline test/E2ETest/Microsoft.Test.E2E.AspNet.OData/Build.AspNetCore/Microsoft.Test.E2E.AspNetCore.OData.csproj:1862]: Should the 4 `<Private>True</Private>` nested elements be retained?

File: src/Microsoft.AspNet.OData/Microsoft.AspNet.OData.csproj (hunk 1 above)
Comment (KenitoInc) [reply src/Microsoft.AspNet.OData/Microsoft.AspNet.OData.csproj:34]: I did not manually delete it. I wonder why it keeps on getting deleted

**Review by xuzhg — APPROVED** [review]

**Review by xuzhg — APPROVED** [review]

**Pull request merged**